python/lsst/ts/standardscripts/maintel/m1m3/check_actuators.py

```diff
@@ -0,0 +1,258 @@
+
+        url = "https://github.com/lsst-ts/"
+        path = (
+            "ts_externalscripts/blob/main/python/lsst/ts/standardscripts/"
+            "maintel/m1m3/check_actuators.py"
+        )
+        schema_yaml = f"""
+        $schema: http://json-schema.org/draft-07/schema#
+        $id: {url}{path}
+        title: CheckAcutators v1
+        description: Configuration for Maintel bump test SAL Script.
+        type: object
+        properties:
+            actuators:
+                description: Actuators to run the bump test.
+                oneOf:
+                  - type: array
+                    items:
+                      type: number
+                      enum: [{m1m3_actuator_ids_str}]
+                    minItems: 1
+                    uniqueItems: true
+                    additionalItems: false
+                  - type: string
+                    enum: ["all"]
+                default: "all"
+        additionalProperties: false
+        """
+        return yaml.safe_load(schema_yaml)
+
+    async def configure(self, config):
+        """Configure the script.
+
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+            Configuration
+        """
+
+        # Getting actuators to be tested.
+        self.actuators_to_test = (
+            self.m1m3_actuator_ids if config.actuators == "all" else config.actuators
+        )
+
+    def set_metadata(self, metadata):
+        """Set metadata."""
+
+        # Getting total number of secondary actuators to be tested
+        total_tested_secondary = sum(
+            [
+                1
+                for actuator in self.actuators_to_test
+                if self.has_secondary_actuator(actuator)
+            ]
+        )
+
+        # Setting metadata
+        metadata.duration = self.time_one_bump * (
+            len(self.actuators_to_test) + total_tested_secondary
+        )
+
+    def has_secondary_actuator(self, actuator_id: int) -> bool:
+        """Determines whether a given actuator has a
+        secondary axis or not.
+        """
+
+        return actuator_id in self.m1m3_secondary_actuator_ids
+
+    async def run(self):
+        await self.assert_feasibility()
+        start_time = time.time()
+
+        # Get M1M3 detailed state
+        detailed_state = DetailedState(
+            (
+                await self.mtcs.rem.mtm1m3.evt_detailedState.aget(
+                    timeout=self.mtcs.fast_timeout,
+                )
+            ).detailedState
+        )
+        self.log.info(f"Current M1M3 detailed state: {detailed_state!r}.")
+
+        # Put M1M3 in engineering mode
+        await self.mtcs.enter_m1m3_engineering_mode()
+
+        failed_actuators = []
+        for i, actuator_id in enumerate(self.actuators_to_test):
+            await self.mtcs.assert_all_enabled()
+            await self.checkpoint(f"Running bump test on FA ID: {actuator_id}.")
```

Comment:
For digging out data from EFD, it's handy to have indices in the log message. Consider something like:
```python
    primary_index = self.m1m3_actuator_ids.index(actuator_id)
    if self.has_secondary(actuator_id):
        secondary_index = self.m1m3_secondary_actuator_ids.index()
        await self.checkpoint(f"Running bump test on DAA FA ID: {actuator_id} (primary {primary_index} secondary {secondary_index})")
    else:
        await self.checkpoint(f"Running bump test on SAA FA ID: {actuator_id} (primary {primary_index})")
```

Reply:
I would prefer to include this information in the final message, along with the bump test results.